python/examples/mote-api.py

```diff
@@ -51,13 +64,134 @@ def mote_on(status):
     mote.show()
     return True
 
+
+## One Light, on One Mote Stick, Travelling at the set colour & channel from the USB port to the End
+def larson_usb_to_end(colour,channel):
+    r, g, b = hex_to_rgb(colour)
+    for i in range(0,16,1):
+        mote.set_pixel(channel,i,r,g,b)
+        mote.show()
+        time.sleep(pause_time_default)
+        mote.clear()
+
+## One Light, on One Mote Stick, Travelling at the set colour & channel from the End to the USB port
+def larson_end_to_usb(colour,channel):
+    r, g, b = hex_to_rgb(colour)
+    for i in range(15,-1,-1):
+        mote.set_pixel(channel,i,r,g,b)
+        mote.show()
+        time.sleep(pause_time_default)
+        mote.clear()
+
+## One Light, on Two Mote sticks at a time, Travelling at the set colour & channels from the USB port to the End
+def larson_double_usb_to_end(colour,ch1,ch2):
+    r, g, b = hex_to_rgb(colour)
+    for i in range(0,16,1):
+        mote.set_pixel(ch1,i,r,g,b)
+        mote.set_pixel(ch2,i,r,g,b)
+        mote.show()
+        time.sleep(pause_time_default)
+        mote.clear()
+
+## One Light, on Two Mote sticks at a time, Travelling at the set colour & channels from the End side to the USB port
+def larson_double_end_to_usb(colour,ch1,ch2):
+    r, g, b = hex_to_rgb(colour)
+    for i in range(15,-1,-1):
+        mote.set_pixel(ch1,i,r,g,b)
+        mote.set_pixel(ch2,i,r,g,b)
+        mote.show()
+        time.sleep(pause_time_default)
+        mote.clear()
+
+## One Light, on Four Mote sticks at a time, Travelling at the set colour & channels from the USB port to the End
+def larson_quad_usb_to_end(colour):
+    r, g, b = hex_to_rgb(colour)
+    for i in range(0,16,1):
+        mote.set_pixel(1,i,r,g,b)
+        mote.set_pixel(2,i,r,g,b)
+        mote.set_pixel(3,i,r,g,b)
+        mote.set_pixel(4,i,r,g,b)
+        mote.show()
+        time.sleep(pause_time_default)
+        mote.clear()
+
+## One Light, on Four Mote sticks at a time, Travelling at the set colour & channels from the End side to the USB port
+def larson_quad_end_to_usb(colour):
+    r, g, b = hex_to_rgb(colour)
+    for i in range(15,-1,-1):
+        mote.set_pixel(1,i,r,g,b)
+        mote.set_pixel(2,i,r,g,b)
+        mote.set_pixel(3,i,r,g,b)
+        mote.set_pixel(4,i,r,g,b)
+        mote.show()
+        time.sleep(pause_time_default)
+        mote.clear()
+
+
+## One Light, remaining On, Travelling at the set colour & channel from the USB port to the End
+def colour_flow_usb_to_end(colour,channel):
+    r, g, b = hex_to_rgb(colour)
+    for i in range(0,16,1):
+        mote.set_pixel(channel,i,r,g,b)
+        mote.show()
+        time.sleep(pause_time_default)
+
+## One Light, remaining On, Travelling at the set colour & channel from the End to the USB port
+def colour_flow_end_to_usb(colour,channel):
+    r, g, b = hex_to_rgb(colour)
+    for i in range(15,-1,-1):
+        mote.set_pixel(channel,i,r,g,b)
+        mote.show()
+        time.sleep(pause_time_default)
```

Comment:
`time.sleep` is blocking, and would cause the web request to hang until the animation effect has completed. Perhaps these effects should be threaded by default- that means they could also run indefinitely- and a new request could stop/change the effect.

I'd suggest a worker thread for this; constantly running in the background to manage the effects and looking at either an effect queue- populated by requests- or at a single `current_effect` value.

Since they also cloud the original purpose of the `mote-api.py` I'd probably also include the effects in a separate Python file (or multiple) and find a way to register them that handles creating the correct routes, etc.

I would likely do this by creating a `MoteEffect` class and deriving each effect from it.